lib/ansible/modules/net_tools/nios/nios_network.py

```diff
@@ -178,6 +178,19 @@ def check_ip_addr_type(ip):
         return NIOS_IPV6_NETWORK
 
 
+def check_vendor_specific_dhcp_option(module, ib_spec):
+    '''This function will check if the argument dhcp option belongs to vendor-specific and if yes then will remove
+     use_options flag which is not supported with vendor-specific dhcp options.
+    '''
+    for key, value in iteritems(ib_spec):
+        if isinstance(module.params[key], list):
```

Comment:
In ib_spec only `options` has a list. What's the need for this check?
Can't you directly use module.params['options']

Reply:
@ganeshrn `options` contains `option_spec` which has the `num` value which I compare for DHCP options so tested it for being `list`. If I do it directly it'll throw `nonetype` error.